store/storage/rocksdb/db.go

```diff
@@ -0,0 +1,271 @@
+package rocksdb
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"math"
+
+	"cosmossdk.io/store/v2"
+	"github.com/linxGnu/grocksdb"
+)
+
+const (
+	TimestampSize = 8
+
+	StorePrefixTpl   = "s/k:%s/"
+	latestVersionKey = "s/latest"
+)
+
+var (
+	_ store.VersionedDatabase = (*Database)(nil)
+
+	defaultWriteOpts = grocksdb.NewDefaultWriteOptions()
+	defaultReadOpts  = grocksdb.NewDefaultReadOptions()
+)
+
+type Database struct {
+	storage  *grocksdb.DB
+	cfHandle *grocksdb.ColumnFamilyHandle
+}
+
+func New(dataDir string) (*Database, error) {
+	storage, cfHandle, err := OpenRocksDB(dataDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open RocksDB: %w", err)
+	}
+
+	return &Database{
+		storage:  storage,
+		cfHandle: cfHandle,
+	}, nil
+}
+
+func NewWithDB(storage *grocksdb.DB, cfHandle *grocksdb.ColumnFamilyHandle) *Database {
+	return &Database{
+		storage:  storage,
+		cfHandle: cfHandle,
+	}
+}
+
+func (db *Database) Close() error {
+	db.storage.Close()
```

Comment:
It just dawned unto me:
Whenever you close these handles, set them to nil so that any future .Close() will fail loudly and prevent any subtle errors so perhaps, please also add:
```go
db.storage = nil
db.cfHandles = nil
```

Reply:
`Close()` should be idempotent IMO

Reply:
> Close() should be idempotent IMO

@alexanderbez by definition, it can't be idempotent given that .Close() has clear side effects of closing and freeing up resources, and especially since that's a handle to CGO resources. That would be the equivalent of letting a double free happen and cause some unintended data/memory corruption, instead of letting a nil panic happen which is much less disastrous and indicative of code that needs to be fixed.